### PR TITLE
Support json predicates in python/pandas connector.

### PR DESCRIPTION
**Pull request by chakankardb**
In this PR, we add support for json predicates in the python connector.
It also adds an integration test.

### CODE REVIEW

File: server/src/test/scala/io/delta/sharing/server/TestDeltaSharingServer.scala [hunk 1]
```diff
@@ -37,6 +37,7 @@ object TestDeltaSharingServer {
     FileUtils.writeStringToFile(pidFile, pid)
     if (sys.env.get("AWS_ACCESS_KEY_ID").exists(_.length > 0)) {
       val serverConfigPath = TestResource.setupTestTables().getCanonicalPath
+      println("Abhijit loading server config from " + serverConfigPath)
```
Comment (linzhou-db) [inline server/src/test/scala/io/delta/sharing/server/TestDeltaSharingServer.scala:40]: remove

File: server/src/universal/conf/delta-sharing-server.yaml.template [hunk 2]
```diff
@@ -56,7 +56,7 @@ stalenessAcceptable: false
 # Whether to evaluate user provided `predicateHints`
 evaluatePredicateHints: false
 # Whether to evaluate user provided `jsonPredicateHints`
-evaluateJsonPredicateHints: false
+evaluateJsonPredicateHints: true
```
Comment (linzhou-db) [inline server/src/universal/conf/delta-sharing-server.yaml.template:59]: To confirm this is intended.

File: python/dev/pytest [hunk 3]
```diff
@@ -17,6 +17,8 @@
 #
 
 PYTHON_EXECUTABLE="${PYTHON_EXECUTABLE:-python3}"
+echo $PYTHON_EXECUTABLE
```
Comment (linzhou-db) [inline python/dev/pytest:20]: keep or remove?

**Review by linzhou-db — APPROVED** [review]
I see the example of jsonPredicateHints here: https://github.com/delta-io/delta-sharing/blob/main/PROTOCOL.md#json-predicates-for-filtering

Wonder if we need comment in the json file to lead the user to the link above.

File: server/src/universal/conf/delta-sharing-server.yaml.template (hunk 2 above)
Comment (chakankardb) [reply server/src/universal/conf/delta-sharing-server.yaml.template:59]: Yes, this is needed to enable the predicates.

File: server/src/test/scala/io/delta/sharing/server/TestDeltaSharingServer.scala (hunk 1 above)
Comment (chakankardb) [reply server/src/test/scala/io/delta/sharing/server/TestDeltaSharingServer.scala]: Done

File: python/dev/pytest (hunk 3 above)
Comment (chakankardb) [reply python/dev/pytest]: Removed.

**Comment by chakankardb** [comment]
> I see the example of jsonPredicateHints here: https://github.com/delta-io/delta-sharing/blob/main/PROTOCOL.md#json-predicates-for-filtering
> 
> Wonder if we need comment in the json file to lead the user to the link above.

Which json file should we add the comment in?

**Comment by linzhou-db** [comment]
> > I see the example of jsonPredicateHints here: https://github.com/delta-io/delta-sharing/blob/main/PROTOCOL.md#json-predicates-for-filtering
> > Wonder if we need comment in the json file to lead the user to the link above.
> 
> Which json file should we add the comment in?

"python file" not "json file"... But I think it's ok to skip since looking for an example in the PROTOCOL page is straightforward.

**Pull request merged**